### PR TITLE
test(extension): add test for analytics tracker

### PR DESCRIPTION
**Pull request by jplorek-atix**
# Checklist

- [ ] JIRA - \<[LW-6791](https://input-output.atlassian.net/browse/LW-6791)>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution
This PR adds unit test for `setOptedInForEnhancedAnalytics` function in onboarding flow and settings section

[LW-6791]: https://input-output.atlassian.net/browse/LW-6791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable import/imports-first */
+const mockMatomoSetChain = jest.fn();
+const mockMatomoSendEvent = jest.fn();
+const mockPostHogSendEvent = jest.fn();
+const mockPostHogSetChain = jest.fn();
+const mockExtendLifespan = jest.fn();
+const mockMakePersisten = jest.fn();
+const mockMakeTemporary = jest.fn();
+const mockGetBackgroundStorage = jest.fn();
```
Comment (DominikGuzei) [inline apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx:10]: We can simplify the mocked user service id like this:
```
/* eslint-disable import/imports-first */
const userIdServiceMock: UserIdService = {
  getId: jest.fn(),
  extendLifespan: jest.fn(),
  makeTemporary: jest.fn(),
  makePersistent: jest.fn(),
  clearId: jest.fn()
};
jest.mock('../getUserIdService', () => ({
  getUserIdService: jest.fn().mockReturnValue(userIdServiceMock)
}));
```

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    </AppSettingsProvider>
+  </BackgroundServiceAPIProvider>
+);
+
+describe('Testing Analytics tracker on SettingsWalletBase component', () => {
+  test('should call makePersisten when enabling analytics', async () => {
+    const { findByTestId } = render(<SettingsSecurityComponentTest />);
+
+    mockMakePersisten.mockReset();
+    mockMakeTemporary.mockReset();
```
[the remark this answers is not in the document]
Comment (jplorek-atix) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx:85]: The issue here is that `setOptedInForEnhancedAnalytics` is call on render in [AnalyticsProvider](https://github.com/input-output-hk/lace/pull/173/files#diff-11122defbe2e4610453f2b1e69f8cc29bd1fac44937b06374634d5b57b81cecaR49), so when the assertions are run, these mocks have already been called once, and what I want to test here is, which function is called when the buttons are clicked, not prior calls

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx (hunk 1 above)
Comment (jplorek-atix) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx:10]: Does it make sense to move this to somewhere else? and make it available for other tests, right now we have three places using this mock. 

[AnalyticsTracker.test.ts](https://github.com/input-output-hk/lace/pull/268/files#diff-c01626bf9f58fdb84a14a38a408152a9b90dc8a5de5a0fabd772a7b69230fcf3)
[WalletSetup.test.tsx](https://github.com/input-output-hk/lace/pull/268/files#diff-3d881f0bdd5b18b6e53ee0fbc083279deb4830cd1a1dbdeaa92b9301998d02f4)
and this file

Comment (marcin-mazurek) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx:10]: Let's do it :)

File: apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/__tests__/WalletSetup.test.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    const { findByTestId } = render(<SetupContainerTest />);
+    const nextAnalyticsAccept = await findByTestId('wallet-setup-step-btn-next');
+    fireEvent.click(nextAnalyticsAccept);
+
+    await waitFor(() => expect(mockMatomoSendEvent).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(mockPostHogSendEvent).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(mockExtendLifespan).toHaveBeenCalledTimes(2)); // this is being called twice, once per tracker, is this correct?
+  });
+
+  test('should call makePersisten when clicking agree', async () => {
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/__tests__/WalletSetup.test.tsx:93]:
```suggestion
  test('should call makePersistent when clicking agree', async () => {
```

File: apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/__tests__/WalletSetup.test.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-magic-numbers */
+/* eslint-disable import/imports-first */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const mockMatomoSetChain = jest.fn();
+const mockMatomoSendEvent = jest.fn();
+const mockPostHogSendEvent = jest.fn();
+const mockPostHogSetChain = jest.fn();
+const mockExtendLifespan = jest.fn();
+const mockMakePersisten = jest.fn();
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/__tests__/WalletSetup.test.tsx:9]:
```suggestion
const mockMakePersistent = jest.fn();
```

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        <AnalyticsProvider>
+          <SettingsSecurity />
+        </AnalyticsProvider>
+      </I18nextProvider>
+    </AppSettingsProvider>
+  </BackgroundServiceAPIProvider>
+);
+
+describe('Testing Analytics tracker on SettingsWalletBase component', () => {
+  test('should call makePersisten when enabling analytics', async () => {
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/src/views/browser-view/features/settings/components/__tests__/SettingsAnalyticsTracker.test.tsx:81]:
```suggestion
  test('should call makePersistent when enabling analytics', async () => {
```

**Review by marcin-mazurek — CHANGES_REQUESTED** [review]
LGTM - left a few minor suggestions :)

**Review by marcin-mazurek — APPROVED** [review]

**Pull request merged**